agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java

```diff
@@ -531,6 +531,15 @@ static void overlayFromEnv(Configuration config, Path baseDir) throws IOExceptio
     overlayProfilerEnvVars(config);
     overlayAadEnvVars(config);
     overlayInstrumentationEnabledEnvVars(config);
+
+    // disable pre-aggregated metric if sampling rate is not 100 and
+    // roleName/connectionString/instrumentationKey overrides are used
+    if ((config.preview.roleNameOverrides.size() != 0
+            || config.preview.connectionStringOverrides.size() != 0
+            || config.preview.instrumentationKeyOverrides.size() != 0)
+        && config.sampling.percentage != 100) {
+      config.internal.preAggregatedMetric.disabled = true;
+    }
```

Comment:
I think this is solved by #2523, let's remove this (but keep the kill switch and the disabled smoke test), and then let's merge and I'll pull the smoke test into #2523 to test there

Reply:
done